turbopack/crates/turbo-tasks-backend/src/backend/mod.rs

```diff
@@ -1671,6 +1671,12 @@ impl<B: BackingStorage> TurboTasksBackendInner<B> {
 
         let has_children = !new_children.is_empty();
 
+        // If the task is not stateful and has no children, it does not have a way to be invalidated
+        // and we can mark it as immutable.
```

Comment:
is there a tests we could write witha  small graph of immutable tasks? i don't know if there is an easy way to query task state from a test...?